weave-js/src/components/PagePanelComponents/Home/Browse3/typeViews/moviepy.video.VideoClip/VideoPlayer.tsx

```diff
@@ -0,0 +1,188 @@
+import {AutoSizer} from 'react-virtualized';
+
+import * as Dialog from '../../../../../../components/Dialog/Dialog';
+import VideoViewer from '../../../../../../components/Panel2/VideoViewer';
+import {LoadingDots} from '../../../../../LoadingDots';
+import {NotApplicable} from '../../NotApplicable';
+import {CustomLink} from '../../pages/common/Links';
+import {useWFHooks} from '../../pages/wfReactInterface/context';
+import {CustomWeaveTypePayload} from '../customWeaveType.types';
+import {CustomWeaveTypeProjectContext} from '../CustomWeaveTypeDispatcher';
+
+type VideoFormat = 'gif' | 'mp4' | 'webm';
+type VideoFileKeys = `video.${VideoFormat}`;
+
+type VideoClipTypePayload = CustomWeaveTypePayload<
+  'moviepy.video.VideoClip.VideoClip',
+  {[K in VideoFileKeys]: string}
+>;
+
+type VideoPlayerProps = {
+  entity: string;
+  project: string;
+  mode?: string;
+  data: VideoClipTypePayload;
+};
+
+const VIDEO_TYPES: Record<VideoFileKeys, VideoFormat> = {
+  'video.gif': 'gif',
+  'video.mp4': 'mp4',
+  'video.webm': 'webm',
+};
+
+const MIME_TYPES: Record<VideoFormat, string> = {
+  gif: 'image/gif',
+  mp4: 'video/mp4',
+  webm: 'video/webm',
+};
+
+export const VideoPlayer: React.FC<VideoPlayerProps> = props => (
+  <AutoSizer style={{height: '100%', width: '100%'}}>
+    {({width, height}) =>
+      width === 0 || height === 0 ? null : (
+        <VideoPlayerWithSize
+          {...props}
+          containerWidth={width}
+          containerHeight={height}
+        />
+      )
+    }
+  </AutoSizer>
+);
+
+type VideoPlayerWithSizeProps = VideoPlayerProps & {
+  containerWidth: number;
+  containerHeight: number;
+};
+
+const VideoPlayerWithSize: React.FC<VideoPlayerWithSizeProps> = ({
+  entity,
+  project,
+  data,
+  containerWidth,
+  containerHeight,
+  mode: propMode,
+}) => {
+  const [showPopup, setShowPopup] = useState(false);
+  const {useFileContent} = useWFHooks();
+  const context = useContext(CustomWeaveTypeProjectContext);
+  const mode = propMode ?? context?.mode;
+
+  // Find the first available video format
+  const videoKey = Object.keys(data.files).find(key => key in VIDEO_TYPES) as
+    | VideoFileKeys
+    | undefined;
+
+  // Always call the hook with consistent arguments
+  const videoBinary = useFileContent(
+    entity,
+    project,
+    videoKey ? data.files[videoKey] : '',
+    {skip: !videoKey}
+  );
+
+  if (!videoKey) {
+    return <NotApplicable />;
+  }
+
+  const fileExt = VIDEO_TYPES[videoKey];
+  const title = data.custom_name || entity.split('-')[0];
+
+  // Link mode (default view)
+  if (!mode || mode !== 'object_viewer') {
+    const videoText = `${fileExt.toUpperCase()} Video`;
+
+    return (
+      <>
+        <div
+          style={{
+            display: 'flex',
+            justifyContent: 'flex-start',
+            width: '100%',
+            margin: 'auto',
+            padding: '6px',
+          }}>
+          <CustomLink text={videoText} onClick={() => setShowPopup(true)} />
```

Comment:
Any chance we could make the link NOT bold? To differentiate from the call name link at the beginning of the row?